File: arlo-admin-server/internal/modules/member/model/member.go
package model

import (
	"time"

	"gorm.io/gorm"
)

// Member C端会员用户模型，对应 member 表
type Member struct {
	ID        uint64         `gorm:"primaryKey" json:"id"`
	Phone     string         `gorm:"uniqueIndex;size:20;not null;default:''" json:"phone"`
	Password  string         `gorm:"size:128;not null;default:''" json:"-"`
	Nickname  string         `gorm:"size:64;not null;default:''" json:"nickname"`
	Avatar    string         `gorm:"size:255;not null;default:''" json:"avatar"`
	Gender    int8           `gorm:"not null;default:0" json:"gender"`
	Openid    string         `gorm:"uniqueIndex;size:64;default:null" json:"openid"`
	Unionid   string         `gorm:"size:64;not null;default:''" json:"unionid"`
	MpOpenid  string         `gorm:"size:64;not null;default:''" json:"mpOpenid"`
	Source    string         `gorm:"size:16;not null;default:'h5'" json:"source"`
	Status    int8           `gorm:"not null;default:1" json:"status"`
	LastLogin *time.Time     `gorm:"" json:"lastLogin"`
	CreatedAt time.Time      `json:"createdAt"`
	UpdatedAt time.Time      `json:"updatedAt"`
	DeletedAt gorm.DeletedAt `gorm:"index" json:"-"`
}

func (Member) TableName() string {
	return "sys_member"
}

func (m *Member) IsEnabled() bool {
	return m.Status == 1
}

File: arlo-admin-web/src/api/modules/member.ts
import request from '../request'
import type { PageData } from '../request'

export interface MemberItem {
  id: number
  phone: string
  nickname: string
  avatar: string
  gender: number
  source: string
  status: number
  lastLogin: string
  createdAt: string
}

export interface MemberDetail extends MemberItem {
  openid: string
  unionid: string
  mpOpenid: string
  updatedAt: string
}

export interface MemberListParams {
  page?: number
  pageSize?: number
  phone?: string
  nickname?: string
  source?: string
  status?: number
}

export interface CreateMemberParams {
  phone: string
  password?: string
  nickname: string
  avatar: string
  gender: number
  source: string
  status: number
}

export interface UpdateMemberParams {
  id: number
  nickname: string
  avatar: string
  gender: number
  source: string
  status: number
}

export interface UpdateMemberPasswordParams {
  id: number
  password: string
}

export function getMemberList(params: MemberListParams) {
  return request.get<PageData<MemberItem>>('/v1/system/member/list', params)
}

export function getMemberDetail(id: number) {
  return request.get<MemberDetail>(`/v1/system/member/${id}`)
}

export function createMember(data: CreateMemberParams) {
  return request.post('/v1/system/member', data)
}

export function updateMember(data: UpdateMemberParams) {
  return request.put('/v1/system/member', data)
}

export function updateMemberPassword(data: UpdateMemberPasswordParams) {
  return request.put('/v1/system/member/password', data)
}

export function updateMemberStatus(id: number, status: number) {
  return request.put(`/v1/system/member/${id}/status`, { status })
}

export function deleteMember(id: number) {
  return request.delete(`/v1/system/member/${id}`)
}
